engine/movie/game_freak_intro.asm

```diff
@@ -0,0 +1,341 @@
+
+SECTION "engine/movie/game_freak_intro.asm", ROMX
+
+GameFreakIntro::
+; Copyright screen and Game Freak logo
+
+	call DisableLCD
+	call ClearVRAM
+
+	ld a, HIGH(vBGMap0)
+	ldh [hBGMapAddress + 1], a
+	xor a
+	ldh [hBGMapAddress], a
+	ldh [hJoyState], a
+	ldh [hSCX], a
+	ldh [hSCY], a
+
+	ld a, $90
+	ldh [hWY], a
+
+	call EnableLCD
+
+	ld c, 10
+	call DelayFrames
+	ld b, SGB_GS_TITLE_SCREEN
+	call GetSGBLayout
+
+	callab IntroCopyRightInfo
+
+	ld a, 1
+	ldh [hBGMapMode], a
+	call WaitBGMap
+	call SetPalettes
+
+	ld c, 3 * 60	; 3 seconds
+	call DelayFrames
+
+	xor a
+	ldh [hWY], a
+	call ClearTileMap
+
+	call .Init
+.loop
+	call .Frame
+	jr nc, .loop
+
+; this was set if user skipped the GF logo by pressing A
+	ld a, [wJumptableIndex]
+	bit 6, a
+	jr nz, .cancelled
+
+; clear carry flag from .PlayFrame
+	and a
+	ret
+
+.cancelled
+	scf
+	ret
+
+.Init:
+	call DisableLCD
+
+	ld hl, vBGMap0
+	ld bc, BG_MAP_WIDTH * SCREEN_HEIGHT
+	xor a
+	call ByteFill
+
+	ld de, GameFreakLogoGFX
+	ld hl, vChars1
+	lb bc, BANK(GameFreakLogoGFX), 28
+	call Get1bpp
+
+	ld hl, GameFreakLogoSparkleGFX
+	ld de, vChars1 tile $1C
+	ld bc, 5 tiles
+	ld a, BANK(GameFreakLogoSparkleGFX)
+	call FarCopyData
+
+	callba InitEffectObject
+
+	ld hl, wSpriteAnimDict
+	ld a, $27	; SPRITE_ANIM_DICT_GS_SPLASH
+	ld [hli], a
+	ld a, $8d
+	ld [hl], a
+
+	xor a
+	ld [wJumptableIndex], a
+	ld [wIntroSceneFrameCounter], a
+	ld [wIntroSceneTimer], a
+	ldh [hSCX], a
+	ldh [hSCY], a
+
+	ld a, 1
+	ldh [hBGMapMode], a
+
+	ld a, SCREEN_HEIGHT_PX
+	ldh [hWY], a
+	call EnableLCD
+
+	ld a, %11100100
+	ld a, %11100100	; redundant
+	ldh [rOBP0], a
+	ld a, %00100100
+	ldh [rOBP1], a
+	ret
+
+
+.Frame:
+; Play one frame of GameFreakPresents sequence.
+; Return carry when the sequence completes or is canceled.
+	ld hl, hJoypadDown
+	ld a, [hl]
+	and %00001111
+	jr nz, .pressed
+
+	ld a, [wJumptableIndex]
+	bit 7, a
+	jr nz, .Finished
+
+	callba EffectObjectJumpNoDelay
+
+	call GameFreakPresentsScene
+	call DelayFrame
+	and a
+	ret
+
+.pressed
+	ld hl, wJumptableIndex
+	set 6, [hl]
+
+.Finished:
+	callab InitEffectObject
+	call ClearTileMap
+	call ClearSprites
+
+	ld c, 16
+	call DelayFrames
+
+	scf
+	ret
+
+
+GameFreakPresentsScene:
+	jumptable .scenes, wJumptableIndex
+
+.scenes
+	dw GameFreakPresents_Wait64Frames
+	dw GameFreakPresents_Star
+	dw GameFreakPresents_PlaceLogo
+	dw GameFreakPresents_LogoSparkles
+	dw GameFreakPresents_PlacePresents
+	dw GameFreakPresents_WaitForTimer
+	dw GameFreakPresents_SetDoneFlag
+
+GameFreakPresents_NextScene:
+	ld hl, wJumptableIndex
+	inc [hl]
+	ret
+
+GameFreakPresents_Wait64Frames:
+	ld c, $40
+	call DelayFrames
+	call GameFreakPresents_NextScene
+	ret
+
+GameFreakPresents_Star:
+	xor a
+	ld [wIntroSceneFrameCounter], a
+	depixel 10, 11, 4, 0
+	ld a, SPRITE_ANIM_INDEX_GS_INTRO_STAR
+	call InitSpriteAnimStruct
+
+	ld hl, $000c	; SPRITEANIMSTRUCT_VAR1
+	add hl, bc
+	ld [hl], $a0	; star path radius
+
+	ld de, $0031	; SFX_GAME_FREAK_LOGO (metronome?)
```

Comment:
`ld de, $0031` should be `ld de, SFX_GAME_FREAK_LOGO` or `SFX_METRONOME`, whichever it's called